boards/arm/frdm_k64f/frdm_k64f_defconfig

```diff
@@ -10,3 +10,6 @@ CONFIG_GPIO=y
 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=120000000
 CONFIG_OSC_EXTERNAL=y
 CONFIG_HAS_DTS=y
+
+# Enable MPU
+CONFIG_HAS_SYSMPU=y
```

Comment:
I think we need to discuss how to approach applications that want to read and write flash in general. Right now, Carbon is already broken with mcuboot for MPU-related reasons (in addition to https://github.com/zephyrproject-rtos/zephyr/pull/122).

Out of tree, we are maintaining patches like this to keep our own builds working (aside from the applications patches):

https://github.com/Linaro/zephyr/commit/14af77b21ce54e10a0c62139c988271f48776b23
https://github.com/Linaro/zephyr/commit/0b2a39cedf5ab1ed4d90efa361b14de89f347e06

I think the MPU support patches that are coming in are awesome -- they've already helped us catch bugs, some of which are in upstream Zephyr. But it would be good to discuss in particular what the plans are for applications which need to read and write flash and access "other" areas of memory like the UID bytes or system memory on STM32.